File: go/problem_5.go
package main

import "fmt"

func gcd(x int, y int) int {
	for y != 0 {
		x, y = y, x%y
	}
	return x
}

func lcm(x int, y int) int {
	return (x * y) / gcd(x, y)
}

func main() {
	n := 1
	for i := 2; i <= 20; i++ {
		n = lcm(n, i)
	}
	fmt.Printf("%d\n", n)
}

File: dart/problem_5.dart
#!/usr/bin/env dart

import 'iterables.dart';

int gcd(int x, int y) => y == 0 ? x : gcd(y, x % y);
int lcm(int x, int y) => (x * y) ~/ gcd(x, y);

void main() {
  print(range(1, 20).reduce(lcm));
}
